### PR TITLE
Fix packaging setup

### PR DESCRIPTION
**Pull request by jonparrott**
* Don't include tests in the binary (wheel) distribution.
* Include tests in the source distribution.
* Remove (seemingly?) unneeded package_dir.
* Use io.open over open for proper encoding.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -43,7 +45,7 @@
     'protobuf>=3.0.0, <4.0dev',
 ]
 
-with open('README.rst', 'r') as readme:
+with io.open('README.rst', 'r') as readme:
```
Comment (lukesneeringer) [inline setup.py]: What is the distinction between `open` and `io.open`?

Comment (jonparrott) [inline setup.py]: `io.open` treats files as text, and is also equivalent to built-in `open` on Python 3.

**Comment by jonparrott** [comment]
> @dhermes recently objected to sending tests to users in the source distribution. What is the final position on this?

Distributing tests in the *source distribution* (sdist) is preferable as it allows downstream packagers to run tests. Distributing tests in the *binary distribution* (wheel) is bad because it ends up being installed in `site-packages`.

**Comment by dhermes** [comment]
I defer to @jonparrott on such things.

**Pull request merged**